include/openthread/platform/border_routing.h

```diff
@@ -96,6 +97,71 @@ extern void otPlatBorderRoutingProcessIcmp6Ra(otInstance *aInstance, const uint8
 extern void otPlatBorderRoutingProcessDhcp6PdPrefix(otInstance                            *aInstance,
                                                     const otBorderRoutingPrefixTableEntry *aPrefixInfo);
 
+/**
+ * IPv4 version number.
+ *
+ * Used to indicate that an IP packet is IPv4 (value 4).
+ */
+static constexpr uint8_t kIpVersion4 = 4;
+
+/**
+ * IPv6 version number.
+ *
+ * Used to indicate that an IP packet is IPv6 (value 6).
+ */
+static constexpr uint8_t kIpVersion6 = 6;
+
+/**
+ * Get the IP version from an IPv4 or IPv6 packet.
+ *
+ * Extracts the version field from the first byte of the IP header.
+ *
+ * @param[in] data A pointer to the start of the IP packet. Must not be NULL.
+ *
+ * @returns The IP version (4 for IPv4, 6 for IPv6).
+ */
+static inline uint8_t  otPlatBorderRoutingGetIpVersion(const uint8_t *data)
+{
+    OT_ASSERT(data != nullptr);
+
+    // Mute compiler warnings.
+    OT_UNUSED_VARIABLE(kIpVersion4);
+    OT_UNUSED_VARIABLE(kIpVersion6);
+
+    return (static_cast<uint8_t>(data[0]) >> 4) & 0x0F;
+}
```

Comment:
Multiple issues here:

* It is a platform `otPlat*` API which you declared and also implemented in a **public header**.
* There should be no inline public/platform APIs.
* The argument format is not correct. All arguments should follow the `a{Name}` format, so it should be `aData`.

Beyond these basic issues:
* Such an API should not be needed at all. But even if we say, "yes, we want this," it is parsing the IP header in a hacky way inside a completely unrelated header file.
* We should not even provide such an API. The platform code should implement and handle such a basic check.